### PR TITLE
Wiki article that references new Zed join docs

### PR DESCRIPTION
**Pull request by philrz**
Since Brim `v0.25.0` won't yet directly use the full Zed Lake API (#1618), the tl;dr on `join` is that the common multi-Pool joins will have to be done from outside the app via `zapi` for now. Now that we have the [`join` docs in the Zed repo](https://github.com/brimdata/zed/tree/main/docs/language/operators#join), the new article added in this PR points to them and walks through an example that uses `zapi` for such a join, after which the data is pushed back into a Pool for further use within Brim.

If you want to see how this doc will look rendered in a GitHub wiki, I've published it at a temporary location at https://github.com/philrz/scratchwiki/wiki.

### CODE REVIEW

**Comment by jameskerr** [comment]
Reads great to me @philrz . It reminds me that the app will need some new designs to take advantage of multiple inputs and from multi line scripts.

**Review by jameskerr — APPROVED** [review]

File: docs/Joining-Data-(v0.25.0+).md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Please report any bugs or usability issues you find when working with cookbooks
+by [opening an issue](https://github.com/brimdata/brim/wiki/Troubleshooting#opening-an-issue)
+or reaching out on the [Brim public Slack](https://www.brimsecurity.com/join-slack/).
+We'd also love to hear your success stories and variations, so please don't be
+shy!
+
+# Example Usage
+
+By its nature, a join operation requires multiple inputs that will
```
Comment (nwt) [inline docs/Joining-Data-(v0.25.0+).md:41]:
```suggestion
By its nature, a join operation requires two inputs that will
```

File: docs/Joining-Data-(v0.25.0+).md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Brim release `v0.25.0` includes initial support for storing data in Zed Lakes.
+However, due to a current limitation ([brim/1618](https://github.com/brimdata/brim/issues/1618)),
+Zed queries issued from within Brim cannot yet access multiple Pools
+simultaneously. Instead such queries are currently limited to accessing the
+data from whichever Pool is currently selected from the **Pools** list.
+
+Because of this limitation, the [streamed input example](https://github.com/brimdata/zed/tree/main/docs/language/operators#example-5---streamed-input)
+is the only one shown that can currently be executed as is from within Brim.
+If the example `fruit.ndjson` and `people.ndjson` was already present in a
```
Comment (nwt) [inline docs/Joining-Data-(v0.25.0+).md:54]:
```suggestion
If the example `fruit.ndjson` and `people.ndjson` are already present in a
```

File: docs/Joining-Data-(v0.25.0+).md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+app), we can see the same query output as shown in the doc.
+
+![Streamed Join Example](media/Join-Streamed.png)
+
+> **Note:** Because the data being displayed is of multiple different shapes,
+> Brim's **Columns** setting has been modified here to set **Headers: On**.
+
+While queries executed from inside Brim are subject to this limitation, the
+Zed backend that's launched behind Brim is not. Therefore if you're willing to
+perform joins at the shell using [`zapi`](https://github.com/brimdata/zed/blob/main/cmd/zed/README.md#zapi)
```
Comment (nwt) [inline docs/Joining-Data-(v0.25.0+).md:65]:
```suggestion
perform joins in the shell using [`zapi`](https://github.com/brimdata/zed/blob/main/cmd/zed/README.md#zapi)
```

File: docs/Joining-Data-(v0.25.0+).md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+To illustrate this, we'll walk through the [example that shows inputs from Pools](https://github.com/brimdata/zed/tree/main/docs/language/operators#example-4---inputs-from-pools).
+To ensure API-compatibility with the Zed backend, we'll use the `zapi` binary
+found in the `zdeps` directory under the Brim [application binaries](https://github.com/brimdata/brim/wiki/Filesystem-Paths#application-binaries-v0250)
+path, specifically on macOS in this case.
+
+Because Brim already takes care of initializing the Zed Lake, we pick up the
+example commands with the creation and population of the separate Pools
+
+```
+$ export ZDEPS="/Applications/Brim.app/Contents/Resources/app.asar.unpacked/zdeps"
```
Comment (nwt) [inline docs/Joining-Data-(v0.25.0+).md:79]: Maybe do `PATH=/Applications/Brim.app/Contents/Resources/app.asar.unpacked/zdeps:$PATH` instead so you don't need the `$ZDEPS/` prefix?

**Review by nwt — APPROVED** [review]

File: docs/Joining-Data-(v0.25.0+).md (hunk 4 above)
Comment (philrz) [reply docs/Joining-Data-(v0.25.0+).md]: @nwt: Yeah, sounds good. I've pushed a commit to set `PATH` instead of the `$ZDEPS/` prefix.

**Pull request merged**